anvil/anvil.go

```diff
@@ -58,22 +60,24 @@ func (a *Anvil) Start(ctx context.Context) error {
 	anvilLog := a.log.New("chain.id", a.cfg.ChainId)
 	anvilLog.Info("starting anvil")
 
-	tempFile, err := os.CreateTemp("", "genesis-*.json")
-	if err != nil {
-		return fmt.Errorf("error creating temporary genesis file: %w", err)
-	}
-
-	_, err = tempFile.Write(a.cfg.Genesis)
-	if err != nil {
-		return fmt.Errorf("error writing to genesis file: %w", err)
-	}
-
 	// Prep args
 	args := []string{
 		"--host", host,
 		"--chain-id", fmt.Sprintf("%d", a.cfg.ChainId),
 		"--port", fmt.Sprintf("%d", a.cfg.Port),
-		"--init", tempFile.Name(),
+	}
+
+	if len(a.cfg.Genesis) > 0 {
+		tempFile, err := os.CreateTemp("", "genesis-*.json")
+		if err != nil {
+			return fmt.Errorf("error creating temporary genesis file: %w", err)
+		}
+		if _, err = tempFile.Write(a.cfg.Genesis); err != nil {
+			return fmt.Errorf("error writing to genesis file: %w", err)
+		}
+
+		defer os.Remove(tempFile.Name())
```

Comment:
After anvil starts, we don't need this tempfile. Can remove immediately when this function returns. It's consumed in memory on anvil startup